### ISSUE TITLE
Panel disappears in view mode after time range change

### ISSUE THREAD

**Issue by raiden-dev** [grafana/grafana#25994]
**TL;DR**

When using panel templating with an existing variable of "query" type with refresh set to "On Time Range Change" in "View" mode of not first (base) panel, the whole panel disappears after changing the time range.

![grafana-issue](https://user-images.githubusercontent.com/1329477/86288416-8ff68900-bbf2-11ea-8f2c-8ddc89a38db9.gif)

Due to the complex nature of the issue, I've made a demo site with the proper environment for easy reproduce.

URL: https://grafana.raiden.dev/d/OAf9sHMMz/dynamic-templating-issue?viewPanel=3&orgId=1&from=now-1h&to=now&var-repeat=All&var-dynamic=60
Login: admin
Password: dynamictemplate

**Detailed description**

We are using a kinda tricky technique to fine-tune and control intervals for bar graphs based on fetching interval value from the InfluxDB source itself using a template variable with `$__to - $__from` expression in a query.

The full query looks like this:

```
SELECT LAST(value) FROM "bar_interval_s" WHERE time < ($__to - $__from)
```

And the corresponding table looks like this:

```
INSERT bar_interval_s value=10 0
INSERT bar_interval_s value=30 3600000
INSERT bar_interval_s value=60 10800000
INSERT bar_interval_s value=120 21600000
INSERT bar_interval_s value=240 43200000
INSERT bar_interval_s value=480 86400000
INSERT bar_interval_s value=960 172800000
INSERT bar_interval_s value=1920 345600000
INSERT bar_interval_s value=3840 691200000
INSERT bar_interval_s value=7680 1382400000
INSERT bar_interval_s value=15360 2764800000
INSERT bar_interval_s value=30720 5529600000
INSERT bar_interval_s value=61440 11059200000
INSERT bar_interval_s value=122880 2211840000
```

Then we are able to use this variable in graph queries and this works just fine in "Dashboard" view with all panels in one screen, but causes this issue in single-panel view.

**Steps to reproduce**:

0. Prepare environment or use demo - https://grafana.raiden.dev
In case of self-setup, you need at least two template variables: one for repeat panels on and one for fetch value from InfluxDB with refresh "On Time Range Change" on the latest.
1. Open the dashboard and choose 2nd panel, then choose "view" (hotkey "v") from the title menu.
2. Try to change the time interval on this screen, e.g., from "Last 1 hour" to "Last 3 hours".

**What happened**:

The panel disappears completely.

**What you expected to happen**:

Changed time range and graph resolution (if set).

**Anything else we need to know?**:

Page reload helps, the issue is only seen after client-side navigation.

We first seen this issue in 6.x releases, upgrade to latest 7.x not helped.

**Environment**:
- Grafana version: v7.0.5 (efbcbb838b)
- Data source type & version: InfluxDB 1.8.0
- OS Grafana is installed on: Docker on Debian Linux 10
- User OS & Browser: Chrome / any
- Grafana plugins: no
- Others: no

**Comment by torkelo** [comment]
Table to replicate: https://play.grafana.org/d/g4_IuHGGk/refresh-on-time-change-and-repeat?viewPanel=4&orgId=1

**Comment by dprokop** [comment]
@torkelo just created a similar setup to the one you have created on play. This feels quite weird tho when I change the time range it all works. Until I change the interval variable value- then the repeated panel(in view mode) disappears. Investigating more.

**Comment by raiden-dev** [comment]
Confirming that the issue was fixed in 7.2.0-beta1, thank you guys!
